contracts/limitOrder/LimitOrderBook.sol

```diff
@@ -219,4 +228,66 @@ contract LimitOrderBook is
 
         return signer;
     }
+
+    //
+    // INTERNAL VIEW
+    //
+
+    function _formatDecimals(
+        uint256 price,
+        uint8 fromDecimals,
+        uint8 toDecimals
+    ) internal pure returns (uint256) {
+        // LOB_ID: Invalid Decimals
+        require(fromDecimals <= toDecimals, "LOB_ID");
+
+        return price.mul(10**(toDecimals.sub(fromDecimals)));
+    }
+
+    function _getPriceByRoundId(address baseToken, uint80 roundId) internal view returns (uint256) {
+        ChainlinkPriceFeed chainlinkPriceFeed = ChainlinkPriceFeed(IBaseToken(baseToken).getPriceFeed());
+        (uint256 price, ) = chainlinkPriceFeed.getRoundData(roundId);
+        return _formatDecimals(price, chainlinkPriceFeed.decimals(), 18);
+    }
+
+    function _verifyTriggerPrice(LimitOrder memory order, uint80 roundIdWhenTriggered) internal view {
+        // NOTE: Chainlink proxy's roundId is always increased
+        // https://docs.chain.link/docs/historical-price-data/
+
+        // LOB_IRI: Invalid RoundId
+        require(order.roundIdWhenCreated > 0 && roundIdWhenTriggered > order.roundIdWhenCreated, "LOB_IRI");
+
+        // LOB_ITP: Invalid Trigger Price
+        require(order.triggerPrice > 0, "LOB_ITP");
+
+        // NOTE: we can only support stop/take-profit limit order for markets that use ChainlinkPriceFeed
+        uint256 triggeredPrice = _getPriceByRoundId(order.baseToken, roundIdWhenTriggered);
+
+        // we need to make sure the price has reached trigger price.
+        // however, we can only know whether index price has reached trigger price,
+        // we didn't know whether market price has reached trigger price
+
+        // rules of advanced order types
+        // https://help.ftx.com/hc/en-us/articles/360031896592-Advanced-Order-Types
+        if (order.orderType == ILimitOrderBook.OrderType.StopLimitOrder) {
+            if (order.isBaseToQuote) {
+                // LOB_SSLOTPNM: Sell Stop Limit Order Trigger Price Not Matched
+                require(triggeredPrice <= order.triggerPrice, "LOB_SSLOTPNM");
+            } else {
+                // LOB_BSLOTPNM: Buy Stop Limit Order Trigger Price Not Matched
+                require(triggeredPrice >= order.triggerPrice, "LOB_BSLOTPNM");
+            }
+        } else if (order.orderType == ILimitOrderBook.OrderType.TakeProfitLimitOrder) {
+            if (order.isBaseToQuote) {
+                // LOB_STLOTPNM: Sell Take-profit Limit Order Trigger Price Not Matched
+                require(triggeredPrice >= order.triggerPrice, "LOB_STLOTPNM");
+            } else {
+                // LOB_BTLOTPNM: Buy Take-profit Limit Order Trigger Price Not Matched
+                require(triggeredPrice <= order.triggerPrice, "LOB_BTLOTPNM");
+            }
+        } else {
+            // LOB_IOT: Invalid Order Type
+            revert("LOB_IOT");
```

Comment:
Perhaps instead of reverting here and adding the condition check at line `124`, what about checking `order.orderType != ILimitOrderBook.OrderType.LimitOrder` inside `_ verifySigner` and return if false?
We can thus save checking for the same thing at two places